enhancements/network/configurable-network-diagnostics-pod-placement.md

```diff
@@ -0,0 +1,289 @@
+all taints. Some cluster administrators require the ability to control the network diagnostics 
+pod placement. For example, due to node-to-node connectivity restrictions, the network-check-source 
+pod has to be scheduled only on a specific subset of nodes.
+
+An additional requirement is to handle a scenario where no nodes are available to deploy the 
+`network-check-source` pod. Instead of the CNO remaining in a `Progressing` state waiting for the 
+deployment to be scheduled, it should detect and handle the use case without affecting the 
+overall operator state.
+
+### User Stories
+
+* As an OpenShift cluster administrator, I want to ensure that the `network-check-source` pod is 
+  only scheduled on an infrastructure node.
+* As an OpenShift cluster administrator, I want to control the set of nodes 
+  `network-check-target` and `network-check-source` pods are scheduled on.
+* As an OpenShift cluster administrator, I want to scale the worker nodes down to zero without 
+  permanently affecting the `cluster/network.operator.openshift.io` conditions.
+  
+### Goals
+
+* Enable OpenShift cluster administrators to control the pod placement of the 
+  `network-check-source` Deployment and the `network-check-target` DaemonSet.
+* Ensure that CNO handles a scenario where there are no available nodes for the 
+  `network-check-source` Deployment without permanently affecting the operator conditions.
+
+### Non-Goals
+
+* Modifying the scope of the connectivity checks performed by the network diagnostics feature.
+* Modifying other properties of the `network-check-source` Deployment and the 
+  `network-check-target` DaemonSet.
+
+## Proposal
+
+This enhancement adds new API fields to the `network.operator.openshift.io` Custom Resource 
+Definition (CRD), that allow administrators to control the pod placement of the 
+`network-check-source` Deployment and the `network-check-target` DaemonSet. During the 
+reconciliation process for the mentioned resources, CNO will configure the relevant 
+`tolerations` and `nodeSelector` fields.  
+Additionally, when there are no nodes that meet the node selector and tolerations criteria for 
+the `network-check-source` Deployment, CNO will emit a warning event instead of putting the operator
+into a constant `Progressing` state. Considering that the network diagnostics feature is only 
+informational it should not affect the `Availabble` and `Degraded` conditions.
+
+### Workflow Description
+
+The OpenShift cluster administrator can configure the pod placement of the 
+`network-check-source` Deployment and the `network-check-target` DaemonSet by changing the `.
+spec.NetworkDiagnostics.SourceDeploymentNodePlacement` and 
+`.spec.NetworkDiagnostics.TargetDaemonsetNodePlacement` fields in the `cluster/network.operator.
+openshift.io` object.  
+
+1. The OpenShift cluster administrator modifies the `network.operator.openshift.io` custom 
+   resource, for example:
+   ```
+   apiVersion: operator.openshift.io/v1
```

Comment:
why use the operator and not the config api?

Reply:
Discussed offline,  I will try to add it to network.config and think about how that is going to work with the disableNetworkDiagnostics that is only in network.operator

Reply:
Done.